Modules/bz2module.c

```diff
@@ -1420,6 +1444,18 @@ BZ2File_init(BZ2FileObject *self, PyObject *args, PyObject *kwargs)
 
     mode = (mode_char == 'r') ? "rb" : "wb";
 
+#ifdef WITH_THREAD
+    if (!self->lock)
```

Comment:
By the way, you can add the following comment:

/* if init is called twice, keep the existing lock */